0_From_Code_First_To_First_Principles/1_code_first/assignment/src/controllers/userController.ts

```diff
@@ -0,0 +1,112 @@
+import { QueryResult } from "pg";
+import { UserData, client, insertUser, updateUser } from "../database";
+
+export const ERROR_VALIDATION = 'ValidationError';
+export const ERROR_USERNAME_TAKEN = 'UsernameAlreadyTaken';
+export const ERROR_EMAIL_TAKEN = 'EmailAlreadyInUse';
+export const ERROR_USER_NOT_FOUND = 'UserNotFound';
+
+const isUsernameTaken = async (username:string):Promise<boolean> => {
```

Comment:
Love it. Functional decomposition. Makin' it work. That's all you gotta do. Very easy to read!